wright/src/grammar/parsers/expression/binary_expression.rs

```diff
@@ -1,8 +1,60 @@
 use crate::grammar::ast::{BinaryExpression, BinaryOp, Expression};
 use crate::grammar::model::{Fragment, HasFragment};
 use crate::grammar::parsers::expression::ToExpression;
+use nom::bytes::complete::tag;
+use nom::character::complete::multispace0;
+use nom::combinator::{recognize, value};
+use nom::error::ErrorKind;
+use nom::multi::fold_many0;
+use nom::sequence::{delimited, pair};
 use nom::IResult;
 
+/// alt() for zero or more parsers (instead of two or more).
+fn alt_single(ops: &[BinaryOp]) -> impl Fn(Fragment) -> IResult<Fragment, BinaryOp> + '_ {
+    move |input| {
+        let mut res = Err(nom::Err::Error((input, ErrorKind::Alt)));
+        for op in ops {
+            res = value(*op, tag(op.token()))(input);
+            if res.is_ok() {
+                break;
+            }
+        }
+        res
+    }
+}
```

Comment:
this function always returns an error when the list is empty? I feel like there perhaps is a better way to do this.

Reply:
Maybe the function can take `(first: BinaryOp, rest: &[BinaryOp])` and then have a macro for varargs calling.